src/hardware/serialport/serialmouse.h

```diff
@@ -22,81 +22,78 @@
-
-	MouseType type   = MouseType::NoMouse; // currently emulated mouse type
-	uint8_t byte_len = 0; // how many bits the emulated mouse transmits in a
-	                      // byte (serial port setting)
-	bool has_3rd_button = false;
-	bool has_wheel      = false;
-	bool port_valid     = false; // false = port settings incompatible with
-	                             // selected mouse
-	uint8_t smooth_div = 1; // time divider value, if > 1 mouse is more
-	                        // smooth than with real HW
-	bool send_ack      = true;
-	uint8_t packet[6]  = {};
-	uint8_t packet_len = 0;
-	uint8_t xmit_idx = UINT8_MAX; // index of byte to send, if >= packet_len
-	                              // it means transmission ended
-	bool xmit_2part = false; // true = packet has a second part, which could
-	                         // not be evaluated yet
-	bool another_move = false; // true = while transmitting a packet we
-	                           // received mouse move event
-	bool another_button = false; // true = while transmitting a packet we
-	                             // received mouse button event
-	uint8_t buttons = 0; // bit 0 = left, bit 1 = right, bit 2 = middle
-	int32_t delta_x = 0; // movement since last transmitted package
-	int32_t delta_y = 0;
-	int32_t delta_w = 0;
+
+    void HandleDeprecatedOptions(CommandLine *cmd);
+    void SetModel(const MouseModelCOM new_type);
+    void AbortPacket();
+    void ClearCounters();
+    void MouseReset();
+    void StartPacketId();
+    void StartPacketData(const bool extended = false);
+    void StartPacketPart2();
+    void SetEventTX();
+    void SetEventRX();
+    void SetEventTHR();
+    uint8_t ClampCounter(const int32_t counter) const;
+
+    const uint8_t  port_id  = 0;
+    const uint16_t port_num = 0; // for logging purposes
+
+    // Mouse model as specified in the parameter
+    MouseModelCOM param_model = MouseModelCOM::NoMouse;
+    // If true = autoswitch between param_model and Mouse Systems mouse
+    bool param_auto_msm = false;
+
+    MouseModelCOM model = MouseModelCOM::NoMouse; // currently emulated model
+
+    uint8_t port_byte_len = 0; // how many bits the port transmits in a byte
+
+    bool has_3rd_button = false;
+    bool has_wheel      = false;
+    float rate_coeff    = 1.0f; // coefficient for boosted sampling rate
+    bool send_ack       = true;
+    uint8_t packet[6]   = {};
+    uint8_t packet_len  = 0;
+    uint8_t xmit_idx = UINT8_MAX; // index of byte to send, if >= packet_len
+                                  // it means transmission ended
+    bool xmit_2part     = false; // true = packet has a second part, which could
+                                 // not be evaluated yet
+    bool another_move   = false; // true = while transmitting a packet we
+                                 // received mouse move event
+    bool another_button = false; // true = while transmitting a packet we
+                                 // received mouse button event
```

Comment:
For the bools that don't start with `has_`, suggest adding prefixing with some boolean-ness adjustectives (needs, is, has, should, etc..)

Because when I see "another_move" and "another_button" (nouns), I think they hold "move" and "button" objects. 

And when I see "xmit_2part" (verb-start), I think it's a function or a function-pointer that will transmit the "2part".

Reply:
Changed all except `send_ack`, as I'm not 100% sure what is the purpose of this variable.